clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx

```diff
@@ -182,6 +218,23 @@ const PrivacyDeclarationFormTab = ({
     setCurrentDeclaration(declarationToEdit);
   };
 
+  const handleOpenDictModal = () => {
```

Comment:
nit: chakraui has a build in modal handler that automatically creates these functions: https://v1.chakra-ui.com/docs/styled-system/utility-hooks/use-disclosure

using that instead might be nicer